### DISCUSSION TITLE
Same SPEI value for different time-series data

### DISCUSSION THREAD

**Discussion by yifanc17** [monocongo/climate_indices#474] [Q&A]
Hi, 

I have met some issues when calculating 12-month SPEI using `climate-indices` package. The input data are global Precipitation and PET dataset from 1994 to 2100. The calibration period used is 1994-2014. The distribution used for analysis is Pearson Type III (`spei_pearson_12`). Most of the regions got a reasonable SPEI time-series value, but for some spots, the SPEI time-series looked fine for the historical period (1994-2022,  with trend and fluctuation etc.), then it was filled with a single value (for example, SPEI=-3.09 from 2025 to 2100). I checked that both precipitation and PET data showed normal time variation. How did the package produce the same SPEI value with different input under this condition? Besides, if I want to compare the average SPEI values from two different time periods, should I use the whole time range or the historical period as calibration period?

Thanks!

**Comment by monocongo** [comment]
This is interesting, in that you're using synthetic data for future values. We should perhaps look into what's happening around future dates, in case this is what's causing this unexpected behavior? Otherwise, I don't have any obvious clues to offer. Limit your data range to the present and see if that fixes things. But this won't do you much good since what you want to see are SPEI values after the present date, right?

**Reply by yifanc17** [reply]
Yes, I used the CESM2 LENS dataset and SSP370 scenario as I want to see SPEI values in the future. The only reason I can think of is that maybe Pearson Type III distribution is not suitable for calculating extreme values, I checked the spot that is filled with single SPEI value in the future, the only 'abnormal' thing is that its future PET is significantly larger than others (please see below figure, the first row is a typical gridcell with normal SPEI trend, the second row is the abnormal gridcell). Also, do you think it's reasonable to use historical period as calibration period to calculate future SPEI? Thanks!

<img width="1023" alt="abnormal" src="https://user-images.githubusercontent.com/55721206/183781452-8c245ee5-1cae-42b7-8ca5-20c38c92ee19.png">
